src/parser/meta_parser.rs

```diff
@@ -178,80 +188,27 @@ fn parse_meta_get_data_value(buf: &[u8]) -> IResult<&[u8], MetaResponse> {
-            // early return if there were no flags passed in
-            if meta_values_array.is_empty() {
-                return Ok((input, MetaResponse::Status(Status::Stored)));
-            }
-
-            // data is empty in this case
-            let meta_value =
-                construct_meta_value_from_flag_array(meta_values_array, None, Some(Status::Stored))
-                    .map_err(|_| nom::Err::Failure(nom::error::Error::new(buf, Fail)))?;
-
-            Ok((input, MetaResponse::Data(Some(vec![meta_value]))))
-        }
-        // match arm for "NS" response
-        MetaResponse::Status(Status::NotStored) => {
-            let (input, meta_values_array) = parse_meta_flag_values_as_slice(input)
-                .map_err(|_| nom::Err::Failure(nom::error::Error::new(buf, Fail)))?;
-            let (input, _) = crlf(input)?; // consume the trailing crlf and leave the buffer empty
-
-            if meta_values_array.is_empty() {
-                return Ok((input, MetaResponse::Status(Status::NotStored)));
-            }
-
-            let meta_value = construct_meta_value_from_flag_array(
-                meta_values_array,
-                None,
-                Some(Status::NotStored),
-            )
-            .map_err(|_| nom::Err::Failure(nom::error::Error::new(buf, Fail)))?;
-
-            Ok((input, MetaResponse::Data(Some(vec![meta_value]))))
-        }
-        // match arm for "EX" response
-        MetaResponse::Status(Status::Exists) => {
-            let (input, meta_values_array) = parse_meta_flag_values_as_slice(input)?;
-            let (input, _) = crlf(input)?; // consume the trailing crlf and leave the buffer empty
-
-            if meta_values_array.is_empty() {
-                return Ok((input, MetaResponse::Status(Status::Exists)));
-            }
-
-            let meta_value =
-                construct_meta_value_from_flag_array(meta_values_array, None, Some(Status::Exists))
-                    .map_err(|_| nom::Err::Failure(nom::error::Error::new(buf, Fail)))?;
-
-            Ok((input, MetaResponse::Data(Some(vec![meta_value]))))
-        }
-        // match arm for "NF" response
-        MetaResponse::Status(Status::NotFound) => {
-            let (input, meta_values_array) = parse_meta_flag_values_as_slice(input)?;
-            let (input, _) = crlf(input)?; // consume the trailing crlf and leave the buffer empty
 
-            if meta_values_array.is_empty() {
-                return Ok((input, MetaResponse::Status(Status::NotFound)));
-            }
+    match status {
+        // match arm for "MN\r\n" response
+        MetaResponse::Status(Status::NoOp) => Ok((input, MetaResponse::Status(Status::NoOp))),
+        // match arm for "HD", "NS", "EX" & "NF" responses
+        MetaResponse::Status(s) => process_meta_response_without_data_payload(input, s),
+        _ => Err(nom::Err::Error(nom::error::Error::new(
+            input,
+            nom::error::ErrorKind::Eof,
+        ))),
+    }
+}
```

Comment:
These can all be combined into one arm because they are semantically the same, only the response code being returned is different.